Grid/Grid.php

```diff
@@ -336,18 +336,18 @@ public function initialize()
 
         $this->setPersistence($config->isPersisted());
 
-        // Route
-        if (null != $config->getRoute()) {
-            $this->setRouteUrl($this->router->generate($config->getRoute()));
-        }
-
         // Route parameters
         $routeParameters = $config->getRouteParameters();
         if (!empty($routeParameters)) {
             foreach ($routeParameters as $parameter => $value) {
                 $this->setRouteParameter($parameter, $value);
             }
         }
+        
+        // Route
+        if (null != $config->getRoute()) {
+            $this->setRouteUrl($this->router->generate($config->getRoute()));
```

Comment:
It seems that a parameter is missing from the original PR 
was `$this->setRouteUrl($this->router->generate($config->getRoute(), $routeParameters));`